tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py

```diff
@@ -203,12 +203,67 @@ def test_automated_recovery_from_failed_nodes_IPI_reactive(
 @ignore_leftovers
 @tier4
 @tier4a
-@aws_based_platform_required
 @ipi_deployment_required
 class TestAutomatedRecoveryFromStoppedNodes(ManageTest):
 
     osd_worker_node = None
     extra_node = False
+    machineset_name = None
+    start_ready_replica_count = None
+
+    def verify_osd_worker_node_in_ready_state(self, nodes):
+        """
+        Verify that the osd worker node is in ready state.
+        """
+        expected_statuses = [constants.NODE_NOT_READY, constants.NODE_READY]
+        node_status = get_node_status(self.osd_worker_node[0])
+        node_name = self.osd_worker_node[0].name
+
+        log.info(f"The status of the node {node_name} is {node_status} ")
+
+        if node_status not in expected_statuses:
+            log.warning(
+                f"The node {node_name} is not in the expected statuses: {expected_statuses}. "
+                f"Trying to restart the node..."
+            )
+            nodes.restart_nodes_by_stop_and_start(
+                nodes=self.osd_worker_node, force=True
+            )
+            return
+
+        if node_status == constants.NODE_NOT_READY:
+            log.info(f"Starting the node {node_name}...")
+            nodes.start_nodes(nodes=self.osd_worker_node, wait=True)
+            log.info(f"Successfully started node {node_name} instance")
+        else:
+            log.info(
+                f"The node {node_name} is already in the expected status {constants.NODE_READY}"
+            )
+
+    def wait_for_current_replica_count_equal_to_start_ready_replica_count(self):
+        """
+        Wait for the current ready replica count to be equal to the ready replica count
+        at the beginning of the test.
+        """
+        log.info(f"start ready replica count = {self.start_ready_replica_count}")
+        timeout = 180
+        log.info(
+            f"Wait {timeout} seconds for the current ready replica count to be equal "
+            f"to the start ready replica count"
+        )
+        sample = TimeoutSampler(
+            timeout=timeout,
+            sleep=10,
+            func=machine.get_ready_replica_count,
+            machine_set=self.machineset_name,
+        )
+        try:
+            sample.wait_for_func_value(value=self.start_ready_replica_count)
+        except TimeoutExpiredError:
+            log.warning(
```

Comment:
raise Exception or
return False(and True for success).

Reply:
I see your point. I think the easiest way is to raise the exception in a case of failure.